controllers/main.go

```diff
@@ -79,10 +80,13 @@ func randToken() string {
 // empty, http.Request.RemoteAddr. See the sample nginx.conf for using the
 // real_ip module to correctly set the X-Real-IP header.
 func getClientIP(r *http.Request) string {
-	getHost := func(ip string) string {
-		if strings.Contains(ip, ":") {
-			parts := strings.Split(ip, ":")
-			return parts[0]
+	// getHost returns the host portion of a string containing either a
+	// host:port formatted name or just a host. An empty string is returned if
+	// net.SplitHostPort returns an error.
+	getHost := func(hostPort string) string {
+		ip, _, err := net.SplitHostPort(hostPort)
+		if err != nil {
+			return ""
```

Comment:
It can be any number of other non-routable IPs and/or proxies.  The other approach, which was implemented by @jolan is to look at the IPs listed in X-Forwarded-For to find one that is not a proxy or non-routable.  That is exactly what [ngx_http_realip_module](http://nginx.org/en/docs/http/ngx_http_realip_module.html) is designed to do robustly under the hood.